src/common/core/librarycore.js

```diff
@@ -0,0 +1,785 @@
+                            moves.push({from: '/' + relid + oldInfo[guid].path, to: '/' + relid + newInfo[guid].path});
+                        }
+                    }
+
+                    for (guid in oldInfo) {
+                        if (!newInfo[guid]) {
+                            removedNodePaths.push('/' + relid + oldInfo[guid].path);
+                        }
+                    }
+
+                    for (i = 0; i < removedNodePaths.length; i += 1) {
+                        removeLibraryRelations(root, removedNodePaths[i]);
+                    }
+
+                    for (i = 0; i < moves.length; i += 1) {
+                        moveLibraryRelations(root, moves[i].from, moves[i].to);
+                    }
+
+                    innerCore.setProperty(root, relid, updatedLibraryRootHash);
+                    root = self.removeChildFromCache(root, relid);
+                    return TASYNC.call(function (newLibraryRoot) {
+                        return TASYNC.call(function (newLibraryNodes) {
+                            var i,
+                                inMeta = self.getMemberPaths(newLibraryRoot, CONSTANTS.META_SET_NAME),
+                                libraryInfoAttribute = libraryInfo,
+                                libraryFCO;
+
+                            newLibraryNodes.shift();
+                            //set the name of the library root
+                            innerCore.setAttribute(newLibraryRoot, 'name', name);
+
+                            //add library_info
+                            libraryInfoAttribute.hash = updatedLibraryRootHash;
+                            innerCore.setAttribute(newLibraryRoot, '_libraryInfo', libraryInfoAttribute);
+
+                            if (newLibraryNodes.length > 0) {
+                                //connect the FCO as base of libraryFCO, but be sure to remove the nullPtr
+
+                                libraryFCO = self.getBaseRoot(newLibraryNodes[0]);
+                                innerCore.deletePointer(libraryFCO, 'base');
+                                innerCore.setBase(libraryFCO, FCO);
+
+                                //adding new nodes to the global META
+                                console.log('NNP', newNodePaths, inMeta);
+                                for (i = 0; i < newLibraryNodes.length; i += 1) {
+                                    if (newNodePaths.indexOf(self.getPath(newLibraryNodes[i])) !== -1 &&
+                                        inMeta.indexOf(self.getPath(newLibraryNodes[i])) !== -1) {
+                                        innerCore.addMember(root, CONSTANTS.META_SET_NAME, newLibraryNodes[i]);
+                                    }
+                                }
+                            }
+
+                            root.libraryRoots[name] = newLibraryRoot;
+
+                            return logs;
+                        }, self.loadSubTree(newLibraryRoot));
```

Comment:
What will happen with a huge library here? Is it necessary to load them all at once?